pingcap/tiflash
# fix the problem that offset in limit query for tiflash system tables doesn't take effect

### What problem does this PR solve?

Issue Number: close #6747 

Problem Summary: When query tiflash system tables from tidb, tidb will send query like `select ... limit offset, 1024`. And the query only stops when the num of return rows is less than `length`. However, the `offset` parameter is not repected by tiflash, and will always return the same result with any `offset` value. So if the system table has more than 1024 rows, the query will hang forever.

### What is changed and how it works?
Port the implementation from https://github.com/ClickHouse/ClickHouse/blob/19.5/dbms/src/DataStreams/LimitBlockInputStream.cpp#L21

### Check List

Tests 

1. deploy a cluster and import data to make the tiflash system tables `dt_segments` with more than 1024 rows;
2. run the query `select count(*) from information_schema.tiflash_segments` and make sure it succeeds;

Side effects


Documentation


### Release note



```release-note
Fix the problem that query on tiflash system tables may hang under large amount of data
```
